apps/api/src/lib/handle-utils.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  AgentHandleSchema,
+  MAX_HANDLE_LENGTH,
+  MIN_HANDLE_LENGTH,
+} from "@/types/index.js";
+
+/**
+ * Utility functions for agent handle generation and validation
+ */
+
+/**
+ * Converts a name to a valid handle format
+ * @param name The display name to convert
+ * @returns A valid handle string
+ * @example
+ * generateHandleFromName("John Doe") // returns "john_doe"
+ * generateHandleFromName("Agent@123!") // returns "agent_123_"
+ * generateHandleFromName("My Cool Agent") // returns "my_cool_agent"
+ * generateHandleFromName("_underscore") // returns "_underscore"
+ */
+export function generateHandleFromName(name: string): string {
```

Comment:
twitter's and Pump.fun rules:
- max 15 chars
- min 3 chars
- alphanumeric with underscores

note: only lowercase handles. else, we'd have to store the "raw" handle with uppercase values...which is a bit messier.